mmstudio/src/main/java/org/micromanager/internal/dialogs/AcqControlDlg.java

```diff
@@ -1073,11 +1073,23 @@ public final synchronized SequenceSettings loadAcqSettingsFromProfile() {
 
    public final void updateGUIContents() {
       SequenceSettings sequenceSettings = acqEng_.getSequenceSettings();
-      updateGUIFromSequenceSettings(sequenceSettings);
+      if (!SwingUtilities.isEventDispatchThread()) {
+         SwingUtilities.invokeLater(() -> {
+            updateGUIFromSequenceSettings(sequenceSettings);
+         });
+      }
+      else {
+         updateGUIFromSequenceSettings(sequenceSettings);
+      }
```

Comment:
Since we're already polishing this patch -- I guess this change is redundant because `updateGUIFromSequenceSettings()` is now written to work on any thread.

Reply:
I am abit confused.  I thought we are talking about how to make updateGUIFromSequenceSettings to work from any thread.   Do you mean the change to updateGUIContents in the original patch?  Yes, I would not change that function.  However, whatever you see fit, let's go ahead and close this.

Reply:
@nicost I was looking at the total diff of this PR (which is now very readable!). In `updateGUIFromSequenceSettings()` there is now (after this PR) the conditional self-`invokeLater` as shown in your comment above. So in the section I commented on above (in `updateGUIContents()`), there is no need for the redundant conditional `invokeLater`.

Reply:
Understood now and fully agree.

Reply:
@marktsuchida Thanks for pointing it out. You are right. The conditional ```invokeLater()``` is redundant in ```updateGUIContents()```.